lib/phonebook/providers/association_provider.dart

```diff
@@ -1,37 +1,18 @@
 import 'package:flutter_riverpod/flutter_riverpod.dart';
-import 'package:myecl/auth/providers/openid_provider.dart';
 import 'package:myecl/phonebook/class/association.dart';
-import 'package:myecl/phonebook/repositories/association_repository.dart';
-import 'package:myecl/tools/providers/single_notifier.dart';
 
-class AssociationNotifier extends SingleNotifier<Association> {
-  final AssociationRepository associationRepository = AssociationRepository();
-  AssociationNotifier({required String token})
-      : super(const AsyncValue.loading()) {
-    associationRepository.setToken(token);
-  }
-
-  Future<AsyncValue<Association>> loadAssociation(String associationId) async {
-    return await load(
-      () async => associationRepository.getAssociation(associationId),
-    );
+class AssociationNotifier extends Notifier<Association> {
+  @override
+  Association build() {
```

Comment:
Is this useful ?

Reply:
Yes it is required by the `Notifier` class

Reply:
Why do you need to use Notifier here ?
We are using StateNotifier everywhere on the project.

Reply:
The riverpod documentation suggests to use `Notifier` instead of `StateNotifier`

Reply:
Oh, interesting
This has been posted after the creation of the project...
Good luck for the refactoring, can't wait to review this one 😄